### PR TITLE
[PF-1544] Upgrade terra-common-lib to java-17

### PR DESCRIPTION
**Pull request by dexamundsen**
- Upgrade terra-common-lib to java-17
- update dependency versions
- Minor cleanup of Gradle files to keep them consistent with other services
- Minor code refactor to address checkstyle violation warnings

### CODE REVIEW

**Comment by dexamundsen** [comment]
FYI - Will check stairway before this is merged

**Review by zloery — APPROVED** [review]

File: build.gradle
Comment (ddietterich) [inline build.gradle:32]: Hmm, are you sure about this? The artifactory libraries are not organized the way you might expect.

File: build.gradle
Comment (ddietterich) [inline build.gradle:69]: Is there some organizing principle behind the changes here? If so, perhaps make a comment explaining that so future maintainers can follow the organization. Reordering makes reviewing the change a bit challenging...

**Review by ddietterich — APPROVED** [review]
LGTM

File: build.gradle
Comment (dexamundsen) [reply build.gradle:69]: Yes, this will be added as repo guidelines in a new template (different PR), adding comments here in the meantime

File: build.gradle
Comment (dexamundsen) [reply build.gradle:32]: this is based on artifacts in https://broadinstitute.jfrog.io/ui/repos/tree/General/libs-release-local
besides both libs-release-local & libs-release point to the same (libs-release-local)

**Pull request merged**
